lib/amazing_print.rb

```diff
@@ -9,6 +9,16 @@
 # AmazingPrint might be loaded implicitly through ~/.irbrc or ~/.pryrc
 # so do nothing for subsequent requires.
 #
+
+# We attempt to load required libraries before we load an extension to prevent
+# Gemfile load order issues.
+def load_and_require(extension, required_library)
+  require(extension) unless defined?(Object.const_get(required_library))
```

Comment:
I'm not sure that amazing print should be trying to load these dependencies. Perhaps we could leverage `autoload`
https://ruby-doc.org/3.2.2/Kernel.html#method-i-autoload

Reply:
I tried using `autoload(:Mongoid, 'amazing_print/ext/mongoid')` with no luck.

Reply:
I tried outputing `autoload?(:Mongoid)` as well and not matter what, it was always `nil`. I think this is because `autoload` is being used by Mongoid itself, and once it runs, it clears it.

Reply:
What do you think of this idea?
https://github.com/amazing-print/amazing_print/pull/110

Reply:
Great!